### PR TITLE
[INIT][web_search_with_and]

### PR DESCRIPTION
**Pull request by andrius-preimantas**
See **openerp**.py

### CODE REVIEW

File: web_search_with_and/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+##############################################################################
+
+{
+    'name': "Force AND on Search Input",
+    'version': '0.1',
+    'author': 'Versada UAB, Odoo Community Association (OCA)',
+    'category': 'web',
+    'website': 'http://www.versada.lt',
+    'description': """
+When searching on same field multiple times Odoo joins queries with OR.
```
Comment (pedrobaeza) [inline web_search_with_and/__openerp__.py]: Please extract this to a README.rst file and use OCA's template: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: web_search_with_and/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+Press Shift key to join them with AND.
```
Comment (pedrobaeza) [inline web_search_with_and/__openerp__.py]: I suppose this works the same when searching for any of the search fields (not only the same). Isn't it?

Comment (andrius-preimantas) [inline web_search_with_and/__openerp__.py]: Well, yeah, but Advanced search would always use AND's even when searching on same field, where searching in a "regular" way on same field Odoo joins it with OR. But I've updated description so this does not matter any more :-)

File: web_search_with_and/README.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Force AND on Search Input
+===========
```
Comment (pedrobaeza) [inline web_search_with_and/README.rst]: Use same length of the title

File: web_search_with_and/README.rst [hunk 4]
```diff
@@ -0,0 +1,46 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Force AND on Search Input
```
Comment (pedrobaeza) [inline web_search_with_and/README.rst]: A better title would be "Use AND conditions on omnibar search"

File: web_search_with_and/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
+##############################################################################
+#
+# OpenERP, Open Source Management Solution
+# Copyright (C) 2015 by UAB Versada (Ltd.) <http://www.versada.lt>
+# and contributors.
+#
+# All Rights Reserved.
```
Comment (pedrobaeza) [inline web_search_with_and/__openerp__.py]: This doesn't apply.

File: web_search_with_and/__openerp__.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': "Force AND on Search Input",
+    'version': '0.1',
```
Comment (pedrobaeza) [inline web_search_with_and/__openerp__.py]: Why not 1.0?

File: web_search_with_and/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    'author': 'Versada UAB, Odoo Community Association (OCA)',
+    'category': 'web',
+    'website': 'http://www.versada.lt',
+    'description': """
+Force AND on Search Input
```
Comment (pedrobaeza) [inline web_search_with_and/__openerp__.py]: You can remove this part, as it's on README file.

**Comment by RoelAdriaans** [comment]
:+1:

Since it's a client browser change, functional tested in
- Firefox 37 on Linux (Ubuntu 14.10) and Windows 7, 
- Chrome 41 on Linux and Windows 7
- IE 11 on Windows 7

File: web_search_with_and/__openerp__.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+# This file is part of OpenERP. The COPYRIGHT file at the top level of
+# this module contains the full copyright notices and license terms.
+
+{
+    'name': "Use AND conditions on omnibar search",
+    'version': '1.0',
+    'author': 'Versada UAB, Odoo Community Association (OCA)',
+    'category': 'web',
+    'website': 'http://www.versada.lt',
+    'description': """
```
Comment (OSguard) [inline web_search_with_and/__openerp__.py]: can remove description if README.rst is their

**Comment by andrius-preimantas** [comment]
@RoelAdriaans-B-informed, nice, thanks for your effort :-)

**Comment by andrius-preimantas** [comment]
Hello,

May I ask if there's any plans in merging this module to the upstream?

Cheers

**Comment by pedrobaeza** [comment]
There's a comment from @OSguard that hasn't been honoured

**Comment by andrius-preimantas** [comment]
@legalsylvain , @pedrobaeza is there anything else what needs to be changed before it can go to upstream? I believe travis fail is not because of this module.

Cheers
Andrius

**Comment by pedrobaeza** [comment]
It seems that python-lxml package is not available in Travis. I don't want to merge until this get solved to have a green branch when merging. Please retry later the test (restarting the Travis job), and if green, please let me know.

**Comment by rafaelbn** [comment]
Rebuilding for testing!

**Comment by rafaelbn** [comment]
cc @janetrp

**Comment by andrius-preimantas** [comment]
Rebased module to latest 8.0 version. Hopefully it's going to make Travis pass.

Cheers
Andrius

**Comment by andrius-preimantas** [comment]
Tests passes
cc @pedrobaeza

**Comment by nicomacr** [comment]
Hi, I am writing from ADHOC. Congratulations on this feature. Its going to be really usefull!!
We are also  looking foward it!!!

**Comment by pedrobaeza** [comment]
It works flawlessly! :+1: 

I merge, and adapt to latest conventions directly in a commit to not delay more this PR. Thanks for the efforts

**Pull request merged**